src/atoms/openFilesState.ts

```diff
@@ -0,0 +1,141 @@
+import { FileEntry } from '@tauri-apps/api/fs';
+import { atom, Getter } from 'jotai';
+
+export type PaneId = 'LEFT' | 'RIGHT';
+export const DEFAULT_PANE: PaneId = 'LEFT';
+type FileId = string;
+
+interface OpenFilesState {
+  files: Record<FileId, FileEntry>; // We need undefined bc of the lookup
+  panes: Record<PaneId, PaneState>; // We know we will only have values (not undefined)
+}
+
+interface PaneState {
+  id: PaneId;
+  openFiles: FileId[];
+  activeFile?: FileId;
+}
+
+export const openFilesAtom = atom<OpenFilesState>({
+  files: {},
+  panes: {
+    LEFT: {
+      id: 'LEFT',
+      openFiles: [],
+      activeFile: undefined,
+    },
+    RIGHT: {
+      id: 'RIGHT',
+      openFiles: [],
+      activeFile: undefined,
+    },
+  },
+});
+
+export const leftPaneAtom = atom((get) => getPane(get, 'LEFT'));
+export const rightPaneAtom = atom((get) => getPane(get, 'RIGHT'));
+
+function getPane(get: Getter, paneId: PaneId) {
+  const { files, panes } = get(openFilesAtom);
+  const pane = panes[paneId];
+  return {
+    id: pane.id,
+    files: Array.from(pane.openFiles)
+      .map((id) => files[id] as FileEntry | undefined)
+      .filter((f) => f) as FileEntry[],
+    active: pane.activeFile ? files[pane.activeFile] : undefined,
+  };
+}
+
+export const openFileInPaneAtom = atom(null, (get, set, update: { pane?: PaneId; file: string | FileEntry }) => {
```

Comment:
Now I see this signature, I feel like we need a different function to "activate" a file, as you initially suggested. Because the update types should be `{ pane?: PaneId; file: FileEntry }` and `{ pane: PaneId; file: string }` respectively for `openFileInPane` and `activateFileInPane`.
I feel having only this function makes it more confusing

Reply:
Ok. I can make two function here. Activate needs the path and open the file.